### PR TITLE
fix: deleting all models of a category

### PR DESCRIPTION
**Pull request by farnazj**
When deleting a workspace, ignore the exception raised for attempting to delete models whose status is already set to deleted.

### CODE REVIEW

File: .github/workflows/main_sleuth-modified.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+name: Build and deploy Python app to Azure Web App - sleuth-modified
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
```
Comment (alonh) [inline .github/workflows/main_sleuth-modified.yml:13]: This deployment should not be part of the PR

File: label_sleuth/config.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-    "first_model_positive_threshold": 20,
-    "changed_element_threshold": 20,
+    "first_model_positive_threshold": 4,
+    "changed_element_threshold": 4,
```
Comment (alonh) [inline label_sleuth/config.json:3]: please keep the original thresholds. thanks :)

File: label_sleuth/orchestrator/orchestrator_api.py [hunk 3]
```diff
@@ -322,7 +322,10 @@ def delete_iteration_model(self, workspace_id, category_name, iteration_index):
     def _delete_category_models(self, workspace_id, category_name):
         workspace = self.orchestrator_state.get_workspace(workspace_id)
         for idx in range(len(workspace.categories[category_name].iterations)):
-            self.delete_iteration_model(workspace_id, category_name, idx)
+            try:
+                self.delete_iteration_model(workspace_id, category_name, idx)
+            except Exception as e:
+                logging.exception(f"error deleting model for iteration '{idx}' in category '{category_name}' from workspace '{workspace_id}'")
```
Comment (alonh) [inline label_sleuth/orchestrator/orchestrator_api.py:328]: this is a working solution, but a better solution would be to check for the model status and only try to delete it if the status is not deleted.
you can get the model status using `workspace.categories[category_id].iterations[idx].model.model_status`
Thanks!

**Review by alonh — CHANGES_REQUESTED** [review]
Thanks for the suggested changes.
See my comments.

File: .github/workflows/main_sleuth-modified.yml (hunk 1 above)
Comment (farnazj) [reply .github/workflows/main_sleuth-modified.yml]: ah sorry I need to deploy from the main branch of the forked repo which I did after I made this PR. I'll close this PR and make another one from a different branch that doesn't have all these extra commits.

File: label_sleuth/config.json (hunk 2 above)
Comment (farnazj) [reply label_sleuth/config.json:3]: same as above
